app/controllers/catalog_controller.rb

```diff
@@ -25,7 +25,7 @@ def self.modified_field
 
     config.show.tile_source_field = :content_metadata_image_iiif_info_ssm
     config.show.partials.insert(1, :openseadragon)
-    config.search_builder_class = Hyrax::CatalogSearchBuilder
+    config.search_builder_class = SearchBuilder
```

Comment:
I'm looking at another project that uses range-limit (the live site is https://digital.library.emory.edu/) and they don't try to override what Hyrax wants.  You might want to just comment this line out like they do and see if things work as expected.

see https://github.com/emory-libraries/dlp-lux/blob/v1.3.5/app/controllers/catalog_controller.rb#L45

Reply:
From what I found, using Hyrax::CatalogSearchBuilder will cause the abnormal display of the blacklight range limit. I have this conclusion based upon the following reasons:

1. Either commenting out or leaving there the line "config.search_builder_class = SearchBuilder", I found the program always loads the SearchBuilder provided by Blacklight_range_limit, instead of Hyrax::CatalogSearchBuilder.

2. When I replaced the Blacklight_range_limit SearchBuilder with Hyrax::CatalogSearchBuilder, the blacklight_range_limit web widget is not loading correctly, missing the sliding bar for year selection.

3. The SearchBuilder provided by Blacklight_range_limit covers all the functions of Hyrax::CatalogSearchBuilder, and furthermore, it provides some other functions which might be specifically for normal display and functioning of the range limit widget.

Therefore, my conclusion is, even though Emory is comment out the line of "config.search_builder_class = SearchBuilder", their code is still using the Blacklight_range_limit SearchBuilder, and Hyrax::CatalogSearchBuilder itself is not working here.